students/WooseokJ/Lesson 8/CircleWJ.py

```diff
@@ -0,0 +1,151 @@
+import io
+import pytest
+import math
+from math import pi
+
+# Basic properties of circle
+class Circle():
+    def __init__(self, radius):
+        self.radius = radius
+
+    @property
+    def diameter(self):
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        return pi * (self.radius**2)
+
+#  alternator constructor that will let the use create a circle directly with the diameter
+    @classmethod
+    def from_diameter(cls, value):
+        return cls(value/2)
+
+    def __str__(self):
+        return "circle with radius: {:.4f}".format(self.radius)
+
+    def __repr__(self):
+        return "circle({})".format(self.radius)
+
+    def __add__(self, other):
+        return self.radius + other.radius
+
+    def __rmul__(self, other):
+        return self.radius * other
+
+    def __mul__(self, other):
+        return self.radius * other
+
+    def __div__(self, value):
+        return self.radius / value
+
+# allow to compre between two circles
+    def __lt__(self, other):
+        if self.radius < other.radius:
+            return True
+        else:

```

Comment:
This _else_ statement is unnecessary, since it follows a return. The code for this method could be simplified to:

```
    def __lt__(self, other):
        if self.radius < other.radius:
            return True
        return False
```